### PR TITLE
[GraphBolt] Add names to `ItemSets` in `_init_all_nodes_set`

### PR DESCRIPTION
**Pull request by Skeleton003**
## Description

1. Add warning to `_init_all_nodes_set` of `OnDiskDataset` when `graph` is None.
2. Add names to `ItemSet`s in `_init_all_nodes_set`.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by frozenbugs** [comment]
The primary goal of this PR is `Add names to ItemSets in _init_all_nodes_set.`, we should reflect this in the title.
If we have to mention the add warning, or add warning is as important as add names, make it 2 PRs.

**Comment by Skeleton003** [comment]
@frozenbugs please have a look on the updated title, and check whether there are any other problems.

File: python/dgl/graphbolt/impl/ondisk_dataset.py [hunk 1]
```diff
@@ -498,13 +499,18 @@ def _init_tvt_set(
...
+            warnings.warn(
+                "Initializing `all_nodes_set` with a graph whose type does "
+                "not belong to SamplingGraph",
+                Warning,
+            )
             return None
         num_nodes = graph.num_nodes
         if isinstance(num_nodes, int):
-            return ItemSet(num_nodes)
+            return ItemSet(num_nodes, names="seed_nodes")
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/impl/ondisk_dataset.py:510]: any dedicated testcases for this change?

Comment (Skeleton003) [reply python/dgl/graphbolt/impl/ondisk_dataset.py:510]: Added just now.

File: python/dgl/graphbolt/impl/ondisk_dataset.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import os
 import shutil
+import warnings
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/impl/ondisk_dataset.py:5]: usually, we use `from ..base import dgl_warning` in DGL.

Comment (Skeleton003) [reply python/dgl/graphbolt/impl/ondisk_dataset.py:5]: Got it.

File: python/dgl/graphbolt/impl/ondisk_dataset.py [hunk 3]
```diff
@@ -498,13 +499,18 @@ def _init_tvt_set(
 
     def _init_all_nodes_set(self, graph) -> Union[ItemSet, ItemSetDict]:
         if graph is None:
+            warnings.warn(
+                "Initializing `all_nodes_set` with a graph whose type does "
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/impl/ondisk_dataset.py:503]: `all_node_set is returned as None as graph is None.`

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**